cryk-ui/src/components/CreateUpdateNewsCardDialog.tsx

```diff
@@ -0,0 +1,172 @@
+  props: TransitionProps & {
+    children: React.ReactElement;
+  },
+  ref: React.Ref<unknown>,
+) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+  
+export default function CreateUpdateNewsCardDialog(props: {operationType: string, dialogQuery: DocumentNode, refetchInput: RefetchInput<GetPaginatedCryptoNewsInput|GetPaginatedCryptocurrenciesInput>, cryptocurrencyId: string, news?: News}) {
+  const [open, setOpen] = React.useState(false);
+
+  const {
+    reset,
+    register,
+    handleSubmit,
+    formState: { isSubmitSuccessful, errors },
+  } = useForm<NewsCardInput>({
+    resolver: zodResolver(newsCardSchema),
+  });
+
+  useEffect(() => {
+    if (isSubmitSuccessful) {
+      reset();
+    }
+  }, [isSubmitSuccessful, reset]);
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const refetchInput = props.refetchInput
+
+  const [ createUpdateNewsEntry ] = useMutation(props.dialogQuery, {
+    context: {clientName: refetchInput.context},
+    refetchQueries: [ {query: refetchInput.query, context: {clientName: refetchInput.context}, variables: refetchInput.variables} ]
+  });
+
+  const onSubmitHandler: SubmitHandler<NewsCardInput> = (values) => {
+    if (props.operationType === "create"){
+      const operationInput: CreateCryptoNewsInput = {title: values.title, body: values.body, about: [ props.cryptocurrencyId ]};
+      createUpdateNewsEntry({
+        variables: {createCryptoNewsInput: operationInput}
+      }).catch((e) => {console.log(JSON.stringify(e, null, 2))});
+    }
+    else {
+      const operationInput: UpdateCryptoNewsInput = {id: (props.news ? props.news.id : ""), title: values.title, body: values.body};
+      createUpdateNewsEntry({
+        variables: {updateCryptoNewsInput: operationInput}
+      }).catch((e) => {console.log(JSON.stringify(e, null, 2))});
+    }
+    setOpen(false);
+  };
+
+  return (
+    <div>
+      <Button 
+        className={props.operationType === "create" ? "news-card-create-button" : "news-card-update-button"}
+        variant="outlined"
+        size={props.operationType === "create" ? "large" : "medium"}
+        onClick={handleClickOpen}
+      >
+        {props.operationType === "create" ? "Create News Card" : "Update"}
+      </Button>
+      <Dialog
+        fullScreen
+        open={open}
+        onClose={handleClose}
+        TransitionComponent={Transition}
+      >
+        <AppBar sx={{ position: "relative" }}>
+          <Toolbar className="toolbar-style">
+            <IconButton
+              edge="start"
+              color="inherit"
+              onClick={handleClose}
+              aria-label="close"
+              size="medium"
+            >
+              <CloseIcon fontSize="large"/>
+            </IconButton>
+            <DialogTitle className="news-dialog-title">
+              {props.operationType === "create" ? "Create News Card" : "Update News Card"}
```

Comment:
Let's create a map containing these values. Afterwards we can refactor these ternary operators to something like this:
```javascript
{textValuesMap[props.operationType]["dialog-title"]}
```